### DISCUSSION TITLE
Active Lineup/Benched

### DISCUSSION THREAD

**Discussion by williamirick** [cwendt94/espn-api#231] [ESPN API] [answered]
First, want to say thank you for all the work that has already been put into this.

Second, I am not sure if I am missing it. Is there anyway to see if a player is currenty benched or in the starting lineup for whatever team? I don't really have any code to show at the moment as I am still exploring/planning.

**Comment by dtcarls** [comment, marked as answer]
To get starters (assuming you have the lineup loaded into the lineup var in this example):
`if lineup.slot_position != 'BE' and lineup.slot_position != 'IR'`

**Reply by williamirick** [reply]
Seems like it is a long work around to get it but beggers cant be choosers! 

Thank you!

**Reply by cwendt94** [reply]
Hey @williamirick, what's the use case you are thinking of? There could be another way of getting this information and I could add some improvements depending on use case. Currently the `slot_position` is set in the [Box Player](https://github.com/cwendt94/espn-api/wiki/Box-Player-Class) which you can get when checking week matchups using `league.box_scores(week)`.

Do you want to look up the player by name and then get this information (team the player is on and line up position) or you thinking of going through all the teams in the league and looking at their roster and player position?

**Reply by williamirick** [reply]
Sorry for the late response, got me on the end of a coding session.

I'm not exactly sure what the end goal is at the moment. I was checking out what exactly is even possible. I'm thinking about trying to put an ESPN fantasy football manager by using PYSimpleGUI.

I'm pretty well-rounded with python but APIs and working with data like this is something im unfamiliar with. 

I'm down to brainstorm ideas on what we could put together.

**Reply by cwendt94** [reply]
No worries!

Interesting, I was digging a little bit into the information I can get from just the player data. It looks like I can move the `slot_position` field to the [Player Class](https://github.com/cwendt94/espn-api/wiki/Player-Class) so that it can always be available even from a teams roster. There is also a field called `onTeamId` which can be added so if you are looking up player information the user could check if the player is a on team.

**Reply by williamirick** [reply]
All I have to say is... I wouldn't complain about any of that! Haha. 

I ran out of time last time, plan on poking around with it more tonight. Somethings that im interested in doing is

Compare the current team to the free agency
- was poking around PYSimpleGUI tables, sorting, selecting, etc.

A more logical team/league home page
- a condensed page of lineups/team activity/league activity/team lineup/team record/playoff odds/etc

A matchup screen once kickoff starts
- Honestly probably not what you intended for but once kickoff happens I don't think there is much that can be added here, at least to start.

Was looking at getting some charts that would be actually useful instead of all those stupid charts that fantasy football platforms tend to place randomly too, haha. 

This post has lost its logical structure but any idea if it is possible to edit lineups from the API? From what I was able to find it did not seem like it was available :(

**Reply by cwendt94** [reply]
Cool, I will create a ticket to add those two new fields to the Player Class!

Nice most of your ideas you can definitely get the information need from this package! The only one that could be tricky is the matchup screen once kickoff starts. A lot of the live updates and data are not part of their API and only used with web sockets.

**Reply by shibedroid** [reply]
Hi, I am a total n00b with python and coding in general. My first project is to extract stats from our fantasy league so I can get interesting pieces of info to my league mates during season. I have learned a lot from your API, thanks a lot!.

Currently I am trying to get two specific things: 

1. Using the Boxscore class, I want to find the highest scoring (active) player for each team each week. I am trying to use **home_lineup** but I noticed that the order of the list does not take into account if the player is in the bench or in the active lineup so what I need to do is to reduce the list or to only include active players for that week and remove the benched players so I can grab the points and sort them

2. Also, I want to add all the points scored by players in the bench (only in the bench). 

I read your previous answers and I think in both cases I could do something with the **slot_position** (**box_score.home_lineup.slot_position**) but I wanted to know if there is already something in place that could allow me to do that with a bit less of work.

EDIT: Also, I searched your code I for the life of me, I don't understand where did **slot_position** come from and how can I know what other "secret" info can I find from a list (home_lineup shows the player name, points and projected points, it does not show the slot position but obviously it is there). Sorry, total n00b here.

Thank you

**Reply by cwendt94** [reply]
Hi @shibedroid, using the `league.box_scores` function you can definitely get those two pieces of data! I have tried to document most of the   package in the repos [wiki page](https://github.com/cwendt94/espn-api/wiki). The data that you see printed in the console is just a preview of the class and is not all of the data you can get.  
Here are a couple references to find more information about each class.
- [Football Overview](https://github.com/cwendt94/espn-api/wiki/Football-Intro)
- [League Interface](https://github.com/cwendt94/espn-api/wiki/League-Class)
- [Box Player Class](https://github.com/cwendt94/espn-api/wiki/Box-Player-Class)
